eclass/node.eclass

```diff
@@ -0,0 +1,49 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EXPORT_FUNCTIONS src_prepare src_compile src_install
+
+RESTRICT+=" mirror"
+SLOT="0"
+
+[ -z "${SRC_URI}" ] && SRC_URI="https://registry.npmjs.org/${PN}/-/${P}.tgz"
+
+[ -z "${NODEJS_BDEPEND}" ] && BDEPEND+=" ${NODEJS_BDEPEND}"
+[ -z "${NODEJS_RDEPEND}" ] && RDEPEND+=" ${NODEJS_RDEPEND}"
+[ -z "${NODEJS_DEPEND}" ] && DEPEND+=" ${NODEJS_DEPEND}"
+[ -z "${RDEPEND}" ] && DEPEND+=" ${RDEPEND}"
+
+DEPEND+=" net-libs/nodejs"
+BDEPEND+="
+	app-misc/jq
+	net-misc/rsync
+	sys-apps/moreutils
+"
+S="${WORKDIR}/package"
+
+node_src_prepare() {
+	#remove version constraints on dependencies
+	jq 'if .dependencies? then .dependencies[] = "*" else . end' package.json | sponge package.json || die
+	default
+}
+
+node_src_compile() {
+	#here we trick npm into believing there are no dependencies so it will not try to fetch them
+	jq 'with_entries(if .key == "dependencies" then .key = "deps" else . end)' package.json | sponge package.json || die
+
+	export npm_config_prefix="${T}/prefix"
+	npm install -g --production || die
+
+	#restore original package.json
+	jq 'with_entries(if .key == "deps" then .key = "dependencies" else . end)' package.json | sponge package.json || die
+}
+
+node_src_install() {
+	#copy files instead of symlinks
+	rsync -avLAX "${T}/prefix/" "${ED}/usr" --exclude /bin || die
+
+	if [ -d "${T}/prefix/bin" ] ; then
```

Comment:
```suggestion
	if [[ -d ${T}/prefix/bin ]] ; then
```
bashisms

Reply:
Who cares? The interpreter will always be bash

Reply:
It matches every other eclass

Reply:
Other eclasses are not even consistent https://github.com/gentoo/gentoo/blob/master/eclass/perl-functions.eclass#L270

Reply:
I reckon it would be if it had been meaningfully edited in the last four years